avocado/plugins/run.py

```diff
@@ -166,7 +168,17 @@ def run(self, args):
             log.error(e.message)
             sys.exit(exit_codes.AVOCADO_FAIL)
         job_instance = job.Job(args)
+
+        # Run JobPre plugins
+        pre_post_dispatcher = JobPrePostDispatcher()
+        output.log_plugin_failures(pre_post_dispatcher.load_failures)
+        pre_post_dispatcher.map_method('pre', job_instance)
+
         job_run = job_instance.run()
+
+        # Run JobPost plugins
+        pre_post_dispatcher.map_method('post', job_instance)
```

Comment:
We do care a lot whether `JobPostTests` is executed when `JobPreTests` was, I'd expect the same for `JobPost`. Can you please put this in `try/finally` block?